### PR TITLE
Valid email address

### PR DESCRIPTION
**Pull request by hiteshvaghasiya83**
- If validate email id then process otherwise show error message

## Description

Put the validation for email id in every place PWA.
- Sing In 
- Create a new account
- Forgot password
- My account edit
- Checkout
  - Guest customer billing address
  - Guest customer order success page

## JIRA

Closes PWA-2361.

## Verification Steps

- When customer Sign In 
- When a customer creates a new account
- When a customer try for forgot the password
- after sign in, Edit Account Info
- Checkout
  - Guest customer billing address
  - Guest customer order success page

#### Test scenario(s) for direct fix/feature

Based on https://en.wikipedia.org/wiki/Email_address#Examples validated email.

**Valid email addresses**

- simple@example.com,
- very.common@example.com,
- disposable.style.email.with+symbol@example.com,
- other.email-with-hyphen@example.com,
- fully-qualified-domain@example.com,
- user.name+tag+sorting@example.com,
- x@example.com,
- example-indeed@strange-example.com,
- test/test@test.com,
- admin@mailserver1,
- example@s.example,
- " "@example.org,
- "john..doe"@example.org,
- mailhost!username@example.org,
- user%example.com@example.org,
- user-@example.org,
- i_like_underscore@but_its_not_allowed_in_this_

**Invalid email addresses**

- Abc.example.com,
- A@b@c@example.com,
- a"b(c)d,e:f;g<h>i[jk]l@example.com,
- just"not"right@example.com,
- this is"notallowed@example.com,
- this still"not\\allowed@example.com,
- 1234567890123456789012345678901234567890123456789012345678901234+x@example.com,
- QA[icon]CHOCOLATE[icon]@test.com

## Related Issue

Facing many issues when login, create an account, and order place due to email id not being validated.

### CODE REVIEW

File: packages/venia-ui/lib/util/formValidators.js
Comment (fooman) [inline packages/venia-ui/lib/util/formValidators.js:170]: Please see here https://hackernoon.com/the-100-correct-way-to-validate-email-addresses-7c4818f24643

suggest to check for the presence of the @ symbol only.

**Comment by larsroettig** [comment]
Hi, @hiteshvaghasiya83 thanks for the contributions pls fix `Conflicting files` also comment from @fooman should be fixed by you. Then I will export to Jira that somebody of core team will review 

best Lars

**Review by larsroettig — CHANGES_REQUESTED** [review]
Hi, @hiteshvaghasiya83 thanks for the contributions pls fix Conflicting files also comment from @fooman should be fixed by you. Then I will export to Jira that somebody of core team will review

best Lars

File: packages/venia-ui/lib/util/formValidators.js
Comment (larsroettig) [inline packages/venia-ui/lib/util/formValidators.js:163]: Should be covered with a test sun you get approval from core member

File: packages/venia-ui/lib/util/formValidators.js
Comment (hiteshvaghasiya83) [reply packages/venia-ui/lib/util/formValidators.js:170]: @fooman given link is not working (https://hackernoon.com/the-100-correct-way-to-validate-email-addresses-7c4818f24643) 
I had created email validation based on the current **Magento 2 Blank/Luma** theme validation, This covers almost all email address validation and This regular expression was validating Magento since Magento 2.0.x.

File: packages/venia-ui/lib/util/formValidators.js
Comment (hiteshvaghasiya83) [reply packages/venia-ui/lib/util/formValidators.js]: @larsroettig, Thank you! I have updated the test case for the email validation function.

File: packages/venia-ui/lib/util/formValidators.js
Comment (fooman) [reply packages/venia-ui/lib/util/formValidators.js:170]: sorry let's see if this works better https://medium.com/hackernoon/the-100-correct-way-to-validate-email-addresses-7c4818f24643

The gist of the article is that emails are so complicated that the best is to check that the user did enter something resembling an email address. Checking for the presence of the @ character achieves that goal. The only way to know for sure that an email address is indeed correct is to have the back-end send a validate your email mail.

Comment (fooman) [reply packages/venia-ui/lib/util/formValidators.js:170]: Probably Luma should be changed as well.

**Comment by supernova-at** [comment]
hi @hiteshvaghasiya83 sorry I lost track of this one, I didn't realize the JIRA ticket automation failed. It's on our board for code review now 👍

**Comment by anthoula** [comment]
Hi @hiteshvaghasiya83 - this looks good. Would you update your branch and resolve conflicting files?

**Review by anthoula — CHANGES_REQUESTED** [review]
Hi @hiteshvaghasiya83 - this looks good. Would you update your branch and resolve conflicting files?

**Comment by supernova-at** [comment]
hey @hiteshvaghasiya83 , sorry for coming through so late in the process but we have previously decided _not_ to do complex email validation on the front end.

See https://github.com/magento/pwa-studio/issues/3094#issuecomment-815879415 for prior discussions of this topic.

That said, we would be okay with a quick "it has an @ symbol" validator if you so desire.

**Comment by supernova-at** [comment]
I'm going to close this for now, please feel free to re-open if you decide to change the validator to check for a `@` only.

**Comment by hiteshvaghasiya83** [comment]
@supernova-at Thank you for the update. Based on https://github.com/magento/pwa-studio/issues/3094#issuecomment-815879415 this comment, I have considered https://medium.com/hackernoon/the-100-correct-way-to-validate-email-addresses-7c4818f24643 for validation and did code likewise I don't know why you close this ticket this code was done by around 2 Oct suddenly team plan to close. Thank you!

**Pull request closed without merging**
